src/Symfony/Component/Ldap/Tests/Security/User/LdapUserProviderTest.php

```diff
@@ -0,0 +1,350 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Tests\Security\User;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Ldap\Adapter\CollectionInterface;
+use Symfony\Component\Ldap\Adapter\QueryInterface;
+use Symfony\Component\Ldap\Entry;
+use Symfony\Component\Ldap\Exception\ConnectionException;
+use Symfony\Component\Ldap\LdapInterface;
+use Symfony\Component\Ldap\Security\User\LdapUser;
+use Symfony\Component\Ldap\Security\User\LdapUserProvider;
+
+/**
+ * @group legacy
+ * @requires extension ldap
+ */
+class LdapUserProviderTest extends TestCase
+{
+    /**
+     * @expectedException \Symfony\Component\Security\Core\Exception\UsernameNotFoundException
+     */
+    public function testLoadUserByUsernameFailsIfCantConnectToLdap()
+    {
+        $ldap = $this->getMockBuilder(LdapInterface::class)->getMock();
```

Comment:
We should use „createMock()“ for „new“ code wherever possible

Reply:
Yes :)